src/main/java/org/folio/rest/impl/PatronBlockLimitsAPI.java

```diff
@@ -0,0 +1,123 @@
+  // IDs come from predefined data (see resources/patron-block-conditions/*)
+  private static final List<String> CONDITIONS_IDS_WITH_DOUBLE_VALUE_TYPE =
+    ImmutableList.of("cf7a0d5f-a327-4ca1-aa9e-dc55ec006b8a");
+
+  @Validate
+  @Override
+  public void getPatronBlockLimits(int offset, int limit, String query, String lang,
+    Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+
+    PgUtil.get(PATRON_BLOCK_LIMITS, PatronBlockLimit.class,
+      org.folio.rest.jaxrs.model.PatronBlockLimits.class, query, offset, limit,
+      okapiHeaders, vertxContext, GetPatronBlockLimitsResponse.class, asyncResultHandler);
+  }
+
+  @Validate
+  @Override
+  public void postPatronBlockLimits(String lang, PatronBlockLimit entity,
+    Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+
+    Errors errors = isEntityValid(entity);
+    if (errors != null) {
+      asyncResultHandler.handle(succeededFuture(PostPatronBlockLimitsResponse
+          .respond422WithApplicationJson(errors)));
+      return;
+    }
+
+    PgUtil.post(PATRON_BLOCK_LIMITS, entity, okapiHeaders, vertxContext,
+      PostPatronBlockLimitsResponse.class, asyncResultHandler);
+  }
+
+  @Validate
+  @Override
+  public void putPatronBlockLimitsByPatronBlockLimitId(String patronBlockLimitId,
+    String lang, PatronBlockLimit entity, Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+
+    Errors errors = isEntityValid(entity);
+    if (errors != null) {
+      asyncResultHandler.handle(succeededFuture(PutPatronBlockLimitsByPatronBlockLimitIdResponse
+          .respond422WithApplicationJson(errors)));
+      return;
+    }
+
+    PgUtil.put(PATRON_BLOCK_LIMITS, entity, patronBlockLimitId, okapiHeaders,
+      vertxContext, PutPatronBlockLimitsByPatronBlockLimitIdResponse.class, asyncResultHandler);
+  }
+
+  @Validate
+  @Override
+  public void getPatronBlockLimitsByPatronBlockLimitId(String patronBlockLimitId,
+    String lang, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+
+    PgUtil.getById(PATRON_BLOCK_LIMITS, PatronBlockLimit.class, patronBlockLimitId,
+      okapiHeaders, vertxContext, GetPatronBlockLimitsByPatronBlockLimitIdResponse.class,
+      asyncResultHandler);
+  }
+
+  @Validate
+  @Override
+  public void deletePatronBlockLimitsByPatronBlockLimitId(String patronBlockLimitId,
+    String lang, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+
+    PgUtil.deleteById(PATRON_BLOCK_LIMITS, patronBlockLimitId, okapiHeaders,
+      vertxContext, DeletePatronBlockLimitsByPatronBlockLimitIdResponse.class,
+      asyncResultHandler);
+  }
+
+  private Errors isEntityValid(PatronBlockLimit entity) {
+
```

Comment:
How about extracting a local variable for `entity.getLimit()` here and passing it to `validateRangeFor*ValueType()` methods? This will save us 4  `entity.getLimit()` calls.

Reply:
done